app/frontend/src/lib/components/Header.svelte

```diff
@@ -0,0 +1,111 @@
+<script lang="ts">
+  import { Refresh } from "$wailsjs/go/main/App"
+  import { deviceLogs, getNextTheme, theme, toggleTheme, type CarbonTheme } from "$lib/state.svelte"
+  import { formatTimestamp } from "$lib/utilities"
+
+  import { Header, HeaderGlobalAction, HeaderUtilities } from "carbon-components-svelte"
+
+  import { RefreshCcw, ToggleLeft } from "@lucide/svelte"
+
+  const themeLabels: Record<CarbonTheme, string> = {
+    g100: "G100",
+    white: "White",
+  }
+
+  const refreshClass = $derived(isRefreshing ? " refresh-action-spinning" : "")
+
+  let lastLog = $derived($deviceLogs?.length ? $deviceLogs[$deviceLogs.length - 1] : undefined)
+  let lastUpdatedTimestamp = $derived(lastLog ? formatTimestamp(lastLog.Time) : formatTimestamp(new Date()))
+  let isRefreshing = $state(false)
+  let refreshResetTimer: ReturnType<typeof setTimeout> | undefined = undefined
+
+  let nextTheme = $derived(getNextTheme($theme))
+  let currentThemeLabel = $derived(themeLabels[$theme])
+  let nextThemeLabel = $derived(themeLabels[nextTheme])
+  let themeTone = $derived($theme === "g100" ? "dark" : "light")
+
+  async function handleRefresh() {
+    if (isRefreshing) {
+      return
+    }
+
+    isRefreshing = true
+    if (refreshResetTimer) {
+      clearTimeout(refreshResetTimer)
+      refreshResetTimer = undefined
+    }
+
+    Refresh()
+
+    refreshResetTimer = setTimeout(() => {
+      isRefreshing = false
+      refreshResetTimer = undefined
+    }, 450)
```

Comment:
Ah, I didn't notice earlier the 450ms here tried to match the 0.45s animation time. Instead of attempting to roughly match the timing in JS which will never work quite right, use https://developer.mozilla.org/en-US/docs/Web/API/Element/animationend_event